unity-renderer/Assets/Scripts/MainScripts/DCL/Controllers/FriendsController/RPCSocialApiBridge.cs

```diff
@@ -189,18 +188,87 @@ public async UniTask RejectFriendshipAsync(string friendRequestId, CancellationT
                         {
                             User = new User() { Address = userId }
                         }
-                    }
+                    },
+                AuthToken = new Payload()
+                {
+                    SynapseToken = accessToken
+                }
             };
 
-            await this.UpdateFriendship(updateFriendshipPayload, cancellationToken);
+            await this.UpdateFriendship(updateFriendshipPayload, userId, cancellationToken);
+        }
+
+        public async UniTask CancelFriendshipAsync(string friendRequestId, CancellationToken cancellationToken = default)
+        {
+            string userId = GetUserIdFromFriendRequestId(friendRequestId);
 
-            OnFriendRequestRemoved?.Invoke(userId);
+            var updateFriendshipPayload = new UpdateFriendshipPayload()
+            {
+                Event =
+                    new FriendshipEventPayload()
+                    {
+                        Cancel = new CancelPayload()
+                        {
+                            User = new User() { Address = userId }
+                        }
+                    },
+                AuthToken = new Payload()
+                {
+                    SynapseToken = accessToken
+                }
+            };
+
+            await this.UpdateFriendship(updateFriendshipPayload, userId, cancellationToken);
         }
 
-        public async UniTask<FriendRequest> RequestFriendshipAsync(string friendUserId, string messageBody, CancellationToken cancellationToken = default)
+        public async UniTask<UserProfile> AcceptFriendshipAsync(string friendRequestId, CancellationToken cancellationToken = default)
```

Comment:
i think it should return a `FriendRequest` instead of the user profile for abstraction purposes

Reply:
I actually found several cohesion issues. Did a major refactor changing many of these cases